src/picongpu/include/fields/FieldB.tpp

```diff
@@ -208,6 +208,23 @@ FieldB::getUnit( )
     return UnitValueType( UNIT_BFIELD, UNIT_BFIELD, UNIT_BFIELD );
 }
 
+HDINLINE
+std::vector<float_64>
+FieldB::getUnitDimension( )
+{
+    /* L, M, T, I, theta, N, J
+     *
+     * B is in Tesla : kg / (A * s^2)
+     *   -> M * T^-2 * I^-1
+     */
+    std::vector<float_64> unitDimension( 7, 0.0 );
+    unitDimension.at(1) =  1.0; // M^1
```

Comment:
only a note: I opened #1431 to allow a more human readable definition that avoids errors with wrong index usage.